process/coordinator/process.go

```diff
@@ -988,25 +985,77 @@ func (tc *transactionCoordinator) CreateMarshalizedReceipts() ([]byte, error) {
 	return tc.marshalizer.Marshal(receiptsBatch)
 }
 
-func (tc *transactionCoordinator) getPostProcessNumOfMbsAndTxs() (int, int) {
-	numMbs := 0
-	numTxs := 0
+func (tc *transactionCoordinator) getNumOfCrossInterMbsAndTxs() (int, int) {
+	totalNumMbs := 0
+	totalNumTxs := 0
 
 	for _, blockType := range tc.keysInterimProcs {
 		interimProc := tc.getInterimProcessor(blockType)
 		if check.IfNil(interimProc) {
 			continue
 		}
 
-		mbs := interimProc.CreateAllInterMiniBlocks()
-		for _, mb := range mbs {
-			numTxs += len(mb.TxHashes)
+		numMbs, numTxs := interimProc.GetNumOfCrossInterMbsAndTxs()
+		totalNumMbs += numMbs
+		totalNumTxs += numTxs
+	}
+
+	return totalNumMbs, totalNumTxs
+}
+
+func (tc *transactionCoordinator) isMaxBlockSizeReached(body *block.Body) bool {
+	numMbs := len(body.MiniBlocks)
+	numTxs := 0
+	numCrossShardScCalls := 0
+
+	allTxs := make(map[string]data.TransactionHandler)
+	for _, blockType := range tc.keysInterimProcs {
+		txs := tc.GetAllCurrentUsedTxs(blockType)
+		for txHash, tx := range txs {
+			allTxs[txHash] = tx
+		}
+	}
+
+	for _, mb := range body.MiniBlocks {
+		numTxs += len(mb.TxHashes)
+
+		isCrossShardTxBlockFromSelf := mb.Type == block.TxBlock &&
+			mb.SenderShardID == tc.shardCoordinator.SelfId() &&
+			mb.ReceiverShardID != tc.shardCoordinator.SelfId()
+
+		if !isCrossShardTxBlockFromSelf {
+			continue
 		}
 
-		numMbs += len(mbs)
+		for txHash := range mb.TxHashes {
```

Comment:
txHash is the index, not the real hash, as seen in some fresh logs

Reply:
:+1: f...g cast to string :)